src/util/index.ts

```diff
@@ -217,27 +223,70 @@ export function create(
     })
   }
 
-  // If the first child is an object then use it to
-  // set attributes.
-  const first = children[0]
-  if (typeof first === 'object' && !(first instanceof Element)) {
-    Object.entries(first).forEach(([key, value]) => {
-      if (value !== undefined) elem.setAttribute(key, value)
+  // If the attrs arg is a Record then use it, otherwise add it to children
+  if (typeof attrs === 'object' && !(attrs instanceof Element)) {
+    Object.entries(attrs).forEach(([key, value]) => {
+      if (value !== undefined) elem.setAttribute(key, `${value}`)
     })
-    children = children.slice(1)
+  } else if (attrs !== undefined) {
+    children = [attrs as typeof children[0], ...children]
```

Comment:
Is this type-cast necessary here? Locally I don't see any errors when removing it.

Reply:
Thanks, indeed was a unnecessary hangover from this function's less type-safe past. Removed.